sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/EpochCoordinator.scala

```diff
@@ -267,4 +268,23 @@ private[continuous] class EpochCoordinator(
       queryWritesStopped = true
       context.reply(())
   }
+
+  private def checkTotalCores(): Unit = {
+    val INSTRUCTION_FOR_FEWER_CORES =
+      """
+        |Total %s (kafka partitions) * %s (cpus per task) = %s needed,
+        |but only have %s (executors) * %s (cores per executor) = %s (total cores).
+        |Please increase total number of executor cores to at least %s.
+    """.stripMargin
+    val numExecutors = session.conf.get("spark.executor.instances", "1").toInt
```

Comment:
This's not always valid for all cluster managers. For example, the number of executors may be decided by the `spark.cores.max` in Standalone.

We can get the number of executors by `BlockManagerMaster.getPeers()`.

Reply:
Thanks for suggestions. Any idea how to get BlockManagerMaster instance here?